File: 2015/01/python/part2.py
import sys

filename = "input.txt"
floor = 0
charCounter = 0

with open(sys.argv[1], 'r') as input_file:
    for i, line in enumerate(input_file):
        for char in line:
            charCounter += 1
            floor = floor + 1 if char == '(' else floor - 1
            if floor == -1:
                print("Santa found the basement after %d tries" % charCounter)
                break

File: 2015/01/golang/part2.go
package adventofcode

import (
	"bufio"
	"fmt"
	"os"
)

func Solve() {
	filename := os.Args[1]
	file, err := os.Open(filename)

	if err != nil {
		fmt.Println("Error encountered")
		return
	}

	fileScanner := bufio.NewScanner(file)
	fileScanner.Split(bufio.ScanLines)

	var floor int
	var tryCounter int

	for fileScanner.Scan() {
		line := fileScanner.Text()
		for i := 0; i < len(line); i++ {
			tryCounter++

			if line[i] == '(' {
				floor += 1
			} else {
				floor -= 1
			}

			if floor == -1 {
				break
			}
		}
	}

	fmt.Printf("Santa found the basement after %d tries", tryCounter)

	file.Close()
}
